### DISCUSSION TITLE
16it database: it works but do not recognize the protection dongle

### DISCUSSION THREAD

**Discussion by das873** [otya128/winevdm#1414] [Q&A]
Hi everyone!
I am new to otdvm/winevdm!

I am trying to migrate an old 16bit database (from Leica Microsystems.. for microscopes...) that I need for working issues from WinXP pc to Win10 pc.
otdvm work perfectly and the database opens correctly!
But the database is proteced by a dongle USB SafeNet Sentinel PRO and it remains always on DEMO mode... 

I used the right driver for the dongle and it works fine under Win10 (I am sure 'cause I have other softwares from Leica, this time 32bit, that work fine with this dongle. The dongle is programmed both for the 16bit database as for the other 32 bit applications)

It seems that the16bit  database, through otdvm, do not have access to dongle's drivers.

Does anyone have idea of what to do?
Thank you so much in advance!!!!

**Comment by cracyc** [comment]
Are you using 32bit windows 10?  If so you can install ntvdm and try that.  If you are using 64bit windows 10 and so have a 64bit dongle driver a trace is probably needed as I have no way of repoing this.

**Comment by das873** [comment]
No... I am using win10 64....

**Comment by das873** [comment]
I suppose the driver could be a 32 bit... It install a sentinel.sys into /system32/drivers.
By the way it works with 32bit applications (both in win10 63 as in winXP... With the same driver). But the 16bit database regognize the dingle only in winXP, but not in Win10 as the database starts with otdvm...

**Comment by das873** [comment]
I was thinking to put the drivers under some folder in otdvm (ex: sin system32 of otdvm)... But. It does not work.
Maybe I need to install the driver under otdvm, but I don't know how...

**Comment by cracyc** [comment]
The driver must be 64bit as a 32bit driver won't work in 64bit windows.  If the 32bit programs work with the dongle then the driver must be installed correctly.  As I said, a trace is needed see https://github.com/otya128/winevdm/issues/226#issuecomment-779702542 for instructions.

**Comment by das873** [comment]
Maybe I have found the problem...
it is a dll from SafeNet Sentinel that is required to unlock the program.
It is the rnbovdd.dll (see attached file).
The error is shown immediately when opening the database...
I tried to copy from a working WinXP system (it is under system32) and t tried to put it both in system32 of Win10 as in syswow64 as well as under the system32 folder of otdvm.
I tried to register the dll with regsvr32.exe, but it reports registering error...

I have no idea how to install this dll that could be used by the database through otdvm...
Any idea?
![Cattura](https://github.com/otya128/winevdm/assets/32270942/2f7bc56e-9aef-4c30-b2cf-b9fc386b900f)

**Comment by cracyc** [comment]
You can try https://github.com/otya128/winevdm/pull/1455 which may work.  I can't actually test it myself due to not having the hardware.

**Comment by das873** [comment]
Thank you so much for your kind reply.
i will surely try!
I'll let you know if it will work!

Best
Roberto

    On Tuesday, October 22, 2024 at 05:13:31 AM GMT+2, cracyc ***@***.***> wrote:  
 
 


You can try #1455 which may work. I can't actually test it myself due to not having the hardware.

**Comment by das873** [comment]
Hi, I am so sorry, but I am not sure what to do with the suggested #1455 files to solve my problem... 😥
May you help me?

**Comment by cracyc** [comment]
Download it from https://ci.appveyor.com/project/otya128/winevdm/builds/50862215/job/y8ghgo8c2xa1ashk/artifacts.

**Reply by das873** [reply]
Thank you so much!
The previous errors are fixed.

But now it reports another error:
![image](https://github.com/user-attachments/assets/9d9d7e29-cc45-4e1e-989f-764760dc5697)

**Comment by cracyc** [comment]
Need a trace.  See https://github.com/otya128/winevdm/issues/226#issuecomment-779702542 for instructions.

**Comment by cracyc** [comment]
https://ci.appveyor.com/project/otya128/winevdm/builds/51381935/job/ee7a18b658ggrlxq/artifacts fixes some vdd bugs.

**Reply by das873** [reply]
Hi,
may you please resent me the fix for vdd bugs?
Many thanks in advance!
Roberto

**Reply by cracyc** [reply]
The newest build contains these fixes https://ci.appveyor.com/project/otya128/winevdm/builds/52940529/job/75khmff32f6loirn/artifacts .
